src/index.ts

```diff
@@ -37,57 +40,111 @@ export class NseIndia {
     private cookieExpiry = new Date().getTime() + (this.cookieMaxAge * 1000)
     private noOfConnections = 0
     private baseHeaders = {
-        'Accept-Language': 'en-US,en;q=0.9',
+        'Accept-Language': 'en-US,en;q=0.5',
         'Accept-Encoding': 'gzip, deflate, br',
         'Connection': 'keep-alive',
-        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0'
+        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/118.0',
     }
 
-    private async getNseCookies() {
-        if (this.cookies === '' || this.cookieUsedCount > 10 || this.cookieExpiry <= new Date().getTime()) {
-            const response = await axios.get(this.baseUrl, {
-                headers: this.baseHeaders
-            })
-            const setCookies = response.headers['set-cookie']
-            const cookies: string[] = []
-            setCookies.forEach((cookie: string) => {
-                const requiredCookies: string[] = ['nsit', 'nseappid', 'ak_bmsc', 'AKA_A2', 'bm_mi', 'bm_sv']
-                const cookieKeyValue = cookie.split(';')[0]
-                const cookieEntry = cookieKeyValue.split('=')
-                /* istanbul ignore else */
-                if (requiredCookies.includes(cookieEntry[0])) {
-                    cookies.push(cookieKeyValue)
+    private async getNseCookies(useSubProcess: boolean = false) {
```

Comment:
We dont need `useSubProcess` variable passed to add the methods. Instead of that, declare one private variable as options, initialised through constructor. And it can be passed while object creation. So it can be used in getData method. We can avoid pass that variable in all the methods.

Reply:
Sure. I will take care from your PR.

Reply:
Thank you.